lib/streamlit/type_util.py

```diff
@@ -653,7 +653,7 @@ def ensure_indexable(obj: OptionSequence[V_co]) -> Sequence[V_co]:
     # function actually does the thing we want.
     index_fn = getattr(it, "index", None)
     if callable(index_fn):
-        return it  # type: ignore[return-value]
+        return copy.copy(it)  # type: ignore[arg-type]
```

Comment:
@vdonato could we add an explanatory comment to this line, since it is not obvious why we do copy here.

Reply:
Ah, yeah good callout -- added